### PR TITLE
[k8s] Add experimental-nvidia-gpus flag when VM has GPU

### PR DESCRIPTION
**Pull request by wbuchwalter**
Mostly for discussion, as I don't think we want custom code like that for every k8s experimental feature.

In order to allow Kubernetes to schedule workloads requiring GPU resource against a node, the 
`--experimental-nvidia-gpus` needs to be set to `1`.
This PR checks the `VMSize` of the agent profile, and set the flag to 1 for any `Standard_N*` size.

Broader related issue: https://github.com/Azure/acs-engine/issues/266

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/267)

### CODE REVIEW

File: pkg/acsengine/engine.go [hunk 1]
```diff
@@ -564,6 +564,14 @@ func (t *TemplateGenerator) getTemplateFuncMap(properties *api.Properties) map[s
 		"HasWindowsSecrets": func() bool {
 			return properties.WindowsProfile.HasSecrets()
 		},
+		"ExperimentalGpuFlag": func() int {
+			//Only test a single profile, since all agents should be the same size
+			isGpuVM := strings.HasPrefix(properties.AgentPoolProfiles[0].VMSize, "Standard_N")
```
Comment (anhowe) [inline pkg/acsengine/engine.go]: what if pool 1 has Standard_N, but pool 0 has Standard_D2_v2?  Will this still enable?

File: pkg/acsengine/engine.go [hunk 2]
```diff
@@ -564,6 +564,14 @@ func (t *TemplateGenerator) getTemplateFuncMap(properties *api.Properties) map[s
...
+			if isGpuVM {
```
Comment (colemickens) [inline pkg/acsengine/engine.go]: Just return the value of `HasPrefix`.

**Comment by anhowe** [comment]
I've unapproved to make sure you turn it on with a GPU capable pool.

to pkg/api/types.go add something like the following (I just confirmed with Drew, the following code will be stable for the next 6-12 months):

func (a *AgentPoolProfile) HasGPU() bool {
	return strings.HasPrefix(a.VMSize, "Standard_N")
}

Then in the template just call "{{.HasGPU}}" instead of "{{ExperimentalGpuFlag}}"

**Comment by wbuchwalter** [comment]
@anhowe Thanks for the feedback, totally missed that.
I may have missed something about how acs-engine works, first time in this code for me, but I also had to modify `engine.go` to allow the variable name to be different for each agent pool.
I didn't find anywhere else where this was done, so I had to add a new syntax `[[VMName]]`.
I'm guessing this might need more discussion to get approved, feel free to point me in another direction or request further changes.

**Comment by brendandburns** [comment]
Can one of the admins verify this patch?

**Comment by wbuchwalter** [comment]
I will close this PR unless someone sees a reason not to.
A new PR will be submitted soon with kubernetes 1.6 support instead.
That will allow us to get rid of the ugly `[[VMName]]` bit, and make the code simpler anyway.

**Comment by wbuchwalter** [comment]
Closing in favor of #385.

**Pull request closed without merging**